tests/test-cases/Group11-Upgrade/11-01-Upgrade.robot

```diff
@@ -173,6 +173,11 @@ Check Container Create Timestamps
 
     # Pause for 2 seconds to allow for the time check below
     Sleep  2
+    # Wait up to a minute extra to allow for time skew between machines
+    Wait Until Keyword Succeeds  30x  2s  Atempt Check Container Create Timestamps  ${newc}
```

Comment:
This output comes from [comparing](https://github.com/docker/cli/blob/a0b19f0ec2a3fe9b0dc4c1aea243b4b0929acb5a/cli/command/formatter/config.go#L91) the time the container was created (according to the container's clock) to the current time (according to the clock on the system running the docker client) and [outputting the difference in a "human-friendly" way](https://github.com/docker/go-units/blob/47565b4f722fb6ceae66b95f853feed578a4a51c/duration.go#L13-L14).

In the case of these test failures, the container's clock is ahead of the clock on the system running the docker client, so the time difference is negative, and "Less than a second" is returned.

Each time we re-check, we're giving the client 2 seconds to advance its clock relative to the creation time (as measured by the container's clock). As long as the two clocks were initially off by less than 1 minute, the client's clock will eventually catch up to the creation time measured by the container's clock.

Because we've configured NTP in vCenter and are using VMware Tools periodic time synchronization (which checks for time skew once a minute by default), 1 minute should be a reasonable buffer.

(If the skew is significant, it may actually take VMware Tools a while to get the clocks entirely synced up, but we don't need to get the clocks synced — we just need the client clock's current time to pass the creation time as recorded by the container VM's clock.)

Reply:
I hadn't considered that! Thanks, that's very helpful.